.github/STYLE_GUIDE.md

```diff
@@ -0,0 +1,344 @@
+and readability of the code.
+
+## Table of Contents
+- [PEP 8](#pep-8)
+- [Line Length](#line-length)
+- [Naming Conventions](#naming-conventions)
+- [Docstrings](#docstrings)
+- [Typing](#typing)
+- [Virtual Environments](#virtual-environments)
+
+<br/>
+
+## PEP 8
+
+Adhere to the PEP 8 style guide, which is the style guide for Python code. Please make sure to familiarize yourself with PEP 8 guidelines: [PEP 8 -- Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/).
+
+<br/>
+
+## Line Length
+
+The maximum line length for code and comments is set to 120 characters. This allows for better readability without excessively long lines.
+
+<br/>
+
+## Naming Conventions
+
+### Classes 
+- Class names should follow the CamelCase convention.
+- Class names should be descriptive and represent a clear concept or object.
+
+    Example:
+
+    ```python
+    class Calculator:
+        def __init__(self):
+            # Constructor implementation
+
+        def add(self, x, y):
+            # Method implementation
+    ```
+
+### Functions
+- Function names should be lowercase, with words separated by underscores.
+- Function names should begin with a verb to indicate the action or operation they perform.
+
+    Examples: 
+
+    ```python
+    def calculate_sum(numbers):
+        """Calculate the sum of a list of numbers."""
+        # Function implementation
+
+    def validate_input(user_input):
+        """Validate user input and return True if valid, False otherwise."""
+        # Function implementation
+
+    def process_data(data):
+        """Process the given data and return the result."""
+        # Function implementation
+    ```
+
+### Variables
+Choosing meaningful and consistent variable names is essential for code readability. Follow these conventions:
+
+- Use lowercase letters with underscores for variable names (snake_case).
+- Be descriptive and use meaningful names to indicate the purpose of the variable.
+
+    Examples:
+
+    ```python
+    # Good variable names
+    user_name = "John"
+    num_items = 5
+    total_amount = 100.50
+
+    # Avoid ambiguous or single-letter names
+    a = "John"  # Not recommended
+    n = 5       # Not recommended
+    ```
+
+
+- Constants should be in uppercase with underscores.
+
+    Examples:
+
+    ```python
+    MAX_RETRIES = 3
+    PI = 3.14159
+    ```
+
+- Avoid using names that shadow built-in functions or keywords.
+
+    Example:
+
+    ```python
+    # Bad: Don't use 'list' as a variable name
+    list = [1, 2, 3]
+
+    # Good: Choose a different name
+    my_list = [1, 2, 3]
+    ```
+
+- Use meaningful prefixes and suffixes for variable names where applicable.
+
+    Example:
+
+    ```python
+    # Prefix 'is_' for boolean variables
+    is_valid = True
+
+    # Suffix iterators with type (such as '_list')
+    name_list = ["John", "Mary", "Robert", "Sue"]
```

Comment:
I'm not sure about this one. I would rather see something like `names` than adding the type as a suffix.

Reply:
The reason to avoid the plural term is because the iterable type is not explicitly clear and could impact how a variable is use later in the code. In this example, "names" could be a tuple or a list or a dict or other. Using a variable like name_list or name_dict keeps usage crystal clear to anyone who needs to consume the variable, without having to refer back to assignment or a type hint.

Reply:
Seems like a use case for type hints to me but maybe that's more of a personal preference :)

Reply:
maybe we should pull the audience and see what the group consensus is. I definitely have personal preference but will yield to the will of the team.